lang/cs/Org.Apache.REEF.IMRU/OnREEF/IMRUTasks/MapTaskHost.cs

```diff
@@ -91,14 +100,46 @@ public byte[] Call(byte[] memento)
              
                 _dataReducer.Send(result);
             }
+            _stopped = true;
             return null;
         }
 
         /// <summary>
+        /// Received close event. 
+        /// If it is closed event sent from driver, set _break to true so that to stop the iteration in Call method
+        /// After _enforceCloseTimeout, if the task is still not closed, throw IMRUTaskSystemException to enforce stop. 
+        /// </summary>
+        /// <param name="closeEvent"></param>
+        public void OnNext(ICloseEvent closeEvent)
+        {
+            if (closeEvent.Value.IsPresent() && Encoding.UTF8.GetString(closeEvent.Value.Value).Equals(TaskManager.CloseTaskByDriver))
+            {
+                Logger.Log(Level.Info, "The task received close event with message: {0}.", Encoding.UTF8.GetString(closeEvent.Value.Value));
+                _break = true;
+            }
+            Thread.Sleep(TimeSpan.FromMilliseconds(_enforceCloseTimeoutMilliseconds));
```

Comment:
After setting the -break to true, we would like Call() to complete the current iteration then return. We should give some time for the the task to gracefully return by itself. But if for whatever reason, blocking call can be one of the reasons, the task is hung and it cannot reach to the next iteration and cannot break by itself, after waiting for specified time, we have to enforce the task to stop. That is the reason we need to sleep here.